diff --git a/homeassistant/components/venstar/__init__.py b/homeassistant/components/venstar/__init__.py
index b543f6d947..c9ed007fda 100644
--- a/homeassistant/components/venstar/__init__.py
+++ b/homeassistant/components/venstar/__init__.py
@@ -25,6 +25,16 @@ from .const import _LOGGER, DOMAIN, VENSTAR_SLEEP, VENSTAR_TIMEOUT
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.CLIMATE, Platform.SENSOR]
 
 
+class VenstarColorTouchWrapper(VenstarColorTouch):
+    @property
+    def name(self) -> str:
+        return "Thermostat"
+
+    @name.setter
+    def name(self, name) -> None:
+        pass
+
+
 async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     """Set up the Venstar thermostat."""
     username = config.data.get(CONF_USERNAME)
@@ -34,7 +44,7 @@ async def async_setup_entry(hass: HomeAssistant, config: ConfigEntry) -> bool:
     timeout = VENSTAR_TIMEOUT
     protocol = "https" if config.data[CONF_SSL] else "http"
 
-    client = VenstarColorTouch(
+    client = VenstarColorTouchWrapper(
         addr=host,
         timeout=timeout,
         user=username,
@@ -77,7 +87,7 @@ class VenstarDataUpdateCoordinator(update_coordinator.DataUpdateCoordinator):
             hass,
             _LOGGER,
             name=DOMAIN,
-            update_interval=timedelta(seconds=60),
+            update_interval=timedelta(seconds=10),
         )
         self.client = venstar_connection
         self.runtimes: list[dict[str, int]] = []
diff --git a/homeassistant/components/venstar/const.py b/homeassistant/components/venstar/const.py
index 31eb572455..cef3a459cb 100644
--- a/homeassistant/components/venstar/const.py
+++ b/homeassistant/components/venstar/const.py
@@ -14,6 +14,6 @@ HOLD_MODE_OFF = "off"
 HOLD_MODE_TEMPERATURE = "temperature"
 
 VENSTAR_TIMEOUT = 5
-VENSTAR_SLEEP = 1.0
+VENSTAR_SLEEP = 0.0
 
 _LOGGER = logging.getLogger(__name__)
