diff --git a/homeassistant/components/venstar/__init__.py b/homeassistant/components/venstar/__init__.py
index ab10ce7ea0b..26d528c360a 100644
--- a/homeassistant/components/venstar/__init__.py
+++ b/homeassistant/components/venstar/__init__.py
@@ -18,6 +18,16 @@ from .coordinator import VenstarConfigEntry, VenstarDataUpdateCoordinator
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.CLIMATE, Platform.SENSOR]
 
 
+class VenstarColorTouchWrapper(VenstarColorTouch):
+    @property
+    def name(self) -> str:
+        return "Thermostat"
+
+    @name.setter
+    def name(self, name) -> None:
+        pass
+
+
 async def async_setup_entry(
     hass: HomeAssistant, config_entry: VenstarConfigEntry
 ) -> bool:
@@ -29,7 +39,7 @@ async def async_setup_entry(
     timeout = VENSTAR_TIMEOUT
     protocol = "https" if config_entry.data[CONF_SSL] else "http"
 
-    client = VenstarColorTouch(
+    client = VenstarColorTouchWrapper(
         addr=host,
         timeout=timeout,
         user=username,
diff --git a/homeassistant/components/venstar/const.py b/homeassistant/components/venstar/const.py
index ea2a7142665..f7a5344e1a1 100644
--- a/homeassistant/components/venstar/const.py
+++ b/homeassistant/components/venstar/const.py
@@ -15,6 +15,6 @@ HOLD_MODE_OFF = "off"
 HOLD_MODE_TEMPERATURE = "temperature"
 
 VENSTAR_TIMEOUT = 5
-VENSTAR_SLEEP = 1.0
+VENSTAR_SLEEP = 0.5
 
 LOGGER = logging.getLogger(__name__)
diff --git a/homeassistant/components/venstar/coordinator.py b/homeassistant/components/venstar/coordinator.py
index 42dc63f0c74..04d14a01835 100644
--- a/homeassistant/components/venstar/coordinator.py
+++ b/homeassistant/components/venstar/coordinator.py
@@ -33,7 +33,7 @@ class VenstarDataUpdateCoordinator(update_coordinator.DataUpdateCoordinator[None
             LOGGER,
             config_entry=config_entry,
             name=DOMAIN,
-            update_interval=timedelta(seconds=60),
+            update_interval=timedelta(seconds=10),
         )
         self.client = venstar_connection
         self.runtimes: list[dict[str, int]] = []
