diff --git a/homeassistant/components/venstar/__init__.py b/homeassistant/components/venstar/__init__.py
index faa47bfc8e4..e33c079e089 100644
--- a/homeassistant/components/venstar/__init__.py
+++ b/homeassistant/components/venstar/__init__.py
@@ -21,6 +21,16 @@ from .coordinator import VenstarDataUpdateCoordinator
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.CLIMATE, Platform.SENSOR]
 
 
+class VenstarColorTouchWrapper(VenstarColorTouch):
+    @property
+    def name(self) -> str:
+        return "Thermostat"
+
+    @name.setter
+    def name(self, name) -> None:
+        pass
+
+
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up the Venstar thermostat."""
     username = config_entry.data.get(CONF_USERNAME)
@@ -30,7 +40,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     timeout = VENSTAR_TIMEOUT
     protocol = "https" if config_entry.data[CONF_SSL] else "http"
 
-    client = VenstarColorTouch(
+    client = VenstarColorTouchWrapper(
         addr=host,
         timeout=timeout,
         user=username,
diff --git a/homeassistant/components/venstar/const.py b/homeassistant/components/venstar/const.py
index a485adad8e7..2f94c1c5d0a 100644
--- a/homeassistant/components/venstar/const.py
+++ b/homeassistant/components/venstar/const.py
@@ -15,6 +15,6 @@ HOLD_MODE_OFF = "off"
 HOLD_MODE_TEMPERATURE = "temperature"
 
 VENSTAR_TIMEOUT = 5
-VENSTAR_SLEEP = 1.0
+VENSTAR_SLEEP = 0.0
 
 _LOGGER = logging.getLogger(__name__)
diff --git a/homeassistant/components/venstar/coordinator.py b/homeassistant/components/venstar/coordinator.py
index 1d0ff60c1e0..fd028920567 100644
--- a/homeassistant/components/venstar/coordinator.py
+++ b/homeassistant/components/venstar/coordinator.py
@@ -32,7 +32,7 @@ class VenstarDataUpdateCoordinator(update_coordinator.DataUpdateCoordinator[None
             _LOGGER,
             config_entry=config_entry,
             name=DOMAIN,
-            update_interval=timedelta(seconds=60),
+            update_interval=timedelta(seconds=10),
         )
         self.client = venstar_connection
         self.runtimes: list[dict[str, int]] = []
